crates/sncast/src/main.rs

```diff
@@ -284,13 +294,25 @@ async fn run_async_command(
 
             let block_id = get_block_id(&block_id)?;
 
+            let class_hash = get_class_hash_by_address(&provider, contract_address)
+                .await?
+                .with_context(|| {
+                    format!(
+                    	"Couldn't retrieve class hash of a contract with address {contract_address:#x}"
+                    )
+                })?;
+
+            let contract_class = get_contract_class(class_hash, &provider).await?;
```

Comment:
Is calldata not optional here? What in case of functions that have no arguments, do we event support these?

Can be a separate issue.